### ISSUE TITLE
Parameter ‘inArray’ shadows a member of 'Level' in writer.h

### ISSUE THREAD

**Issue by GoogleCodeExporter** [byzhang/rapidjson#28]
```
GCC complains about line 92 in writer.h

Level(bool inArray) : inArray(inArray), valueCount(0) {}

Please rename the parameter.

```

Original issue reported on code.google.com by `mte...@googlemail.com` on 12 Jun 2012 at 10:11

**Comment by GoogleCodeExporter** [comment]
```
It is perfectly legal in initializer lists. The problem is on GCC side. 
http://stackoverflow.com/questions/2227244/what-if-a-constructor-parameter-has-t
he-same-name-as-a-member-variable-in-c
```

Original comment by `Mapholameth` on 8 Oct 2012 at 6:54

**Comment by GoogleCodeExporter** [comment]
Original comment by `milo...@gmail.com` on 12 Nov 2012 at 4:02
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision r78.
```

Original comment by `milo...@gmail.com` on 14 Nov 2012 at 2:58
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision r79.
```

Original comment by `milo...@gmail.com` on 14 Nov 2012 at 2:59

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 94d05da2bc0f.
```

Original comment by `milo...@gmail.com` on 6 Jun 2014 at 6:04
